pkg/virt-api/webhooks/username/checker.go

```diff
@@ -0,0 +1,43 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package username
+
+import "fmt"
+
+type Checker struct {
```

Comment:
This new object seems to wrap the service-accounts data.

The data itself is a set (`map[string]struct{}`).

What I do not understand is the reasoning to wrap it using an object.
What added value you see in this, compared to something simple like this:
```go
func KubevirServiceAccounts(namespace string) map[string]struct{} {
  prefix := fmt.Sprintf("system:serviceaccount:%s", kubeVirtNamespace)
  return map[string]struct{}{
	  fmt.Sprintf("%s:%s", prefix, "kubevirt-apiserver"):  {},
	  fmt.Sprintf("%s:%s", prefix, "kubevirt-controller"): {},
	  fmt.Sprintf("%s:%s", prefix, "kubevirt-handler"):    {},
  }
)

func IsKubeVirtServiceAccount(username string, serviceAccounts map[string]struct{}) bool {
  _, ok := serviceAccounts[username]
  return ok
}
```

Reply:
There are two advantages I can see:
1. The map is only populated once.
2. The callers to `IsKubeVirtServiceAccount` do not have to concern themselves about the internal implementation  as is shown in the alternative implementation (passing the map).

BTW, we could also define an alias for the `map[string]struct{}` type and have a method attached to it (in order for not using a struct).

Reply:
>     1. The map is only populated once.

The data is created once be it an object or pure data (as a set).

>     2. The callers to `IsKubeVirtServiceAccount` do not have to concern themselves about the internal implementation  as is shown in the alternative implementation (passing the map).

Running a method vs running a function, in both cases one needs to know what to run and is not aware how it is implemented. You need the data and you need the method/function in both.

> BTW, we could also define an alias for the `map[string]struct{}` type and have a method attached to it (in order for not using a struct).

Yes, it will require less boilerplate but it is still hard to reason for IMO.
But it will be clearer that the difference is in how it is called, i.e. method vs function.